cmd/server/app/migrate_up.go

```diff
@@ -126,10 +130,94 @@ var upCmd = &cobra.Command{
 			return fmt.Errorf("error preparing authz client: %w", err)
 		}
 
+		cmd.Println("Ensuring 'Minder Root' project is provisioned.")
+		store := db.NewStore(dbConn)
+		rootProject, err := ensureRootProject(ctx, cmd, store)
+		if err != nil {
+			return fmt.Errorf("error while ensuring root project: %w", err)
+		}
+
+		cmd.Println("Migrating projects to have root as parent...")
+		projects, err := store.ListNonOrgProjects(ctx)
+		if err != nil {
+			return fmt.Errorf("error while listing non-org projects: %w", err)
+		}
+
+		for _, p := range projects {
```

Comment:
I'd move this for loop into a separate function, perhaps named `addRootToProjects`. As it stands, you are adding a lot of code/logic to an already long function.